### ISSUE TITLE
improved resource support for rhvoice for nvda

### ISSUE THREAD

**Issue by fernando-jose-silva** [RHVoice/RHVoice#915]
Please apologize if I am in the wrong place or if the request is not correct.

I always like to say that I am not a programmer, I am interested in new resources and I am a computer and cell phone instructor in Brazil.

I have two questions for your consideration:

1. When using rh voice in nvda when reading a text with nvda + a keyboard layout for notebooks, if at the beginning of the text there are several empty lines, say 20, the reading does not start. I suspect the problem is with rh voice because this does not happen with other synthesizers like one core.

2. A new add-on for nvda has been released:

Earcons and speech rules;

When using it with the rh voice synthesizer I noticed some errors such as:

If you check the option in nvda Character descriptions with delay when moving the cursor
nvda spells out the description of the characters, for example reading a l f a, expecting to read alpha.

When using nvda in focus mode with the rh voice add-on and synthesizer, when in focus mode nvda is silent and does not read the names of the controls.
These problems do not occur if you use the add-on and another voice as one core.
The author of the add-on
Earcons and speech rules;
informs that the reason rh voice is not compatible is due to:
text from Juda:
RHVoice: The Break command has not been supported since January 2020.
Is it possible to implement this feature in the rh voice add-on for nvda?
Remembering that I do not know how to test functions that are not compiled, (I have no idea how to compile, lol).
Thank you very much for your attention.

**Comment by rmcpantoja** [comment]
Hi.
This previous weekend I have been investigating about **BreakCommand** inclusion, and the resources of the c++ core are limited, because only accepts a value named strength, but not a time in MS, required by the SynthDriver. [see the reference code](https://github.com/RHVoice/RHVoice/blob/master/src/include/core/ssml.hpp)

Other approach was feeding silent audio (in MS) to the end of speech data through the Python SynthDriver, simulating a break. But not sure if it is a good idea because would put performance at risk a bit.

Cheers.

**Comment by zstanecic** [comment]
@rmcpantoja this will be a performance inpact, as you said. People tend to cut silences, not to add more.
Even the some chinese guy wants to improve the perormance by trimming silences, but it is not a topic of this discussion.

**Comment by beqabeqa473** [comment]
Hello.

I've implemented break command extension to include time in pr #994 

Lets wait for further notice by first-party developer.

**Comment by zstanecic** [comment]
#994 is successfully merged. Closing as resolved.

**Comment by fernando-jose-silva** [comment]
Okay, thank you very much. Is there anywhere I can download Adon for NVDa, already compiled and compatible with NVDa version 2026.1? Sorry, please be patient, I'm not a programmer.

**Comment by zstanecic** [comment]
Hi @fernando-jose-silva, I will notify you about the compiled add-on availability. You will be able to get it through NVDA add-on store.

**Comment by fernando-jose-silva** [comment]
Great, thank you very much, looking forward to it.Enviado do meu iPhoneEm 18 de nov. de 2025, à(s) 17:55, ***@***.***> escreveu:﻿zstanecic left a comment (RHVoice/RHVoice#915)
Hi @fernando-jose-silva, I will notify you about the compiled add-on availability. You will be able to get it through NVDA add-on store.
